packages/client/tests/functional/relationMode-m-n-mongodb-failing-with-at-map/tests_m-to-n-MongoDB.ts

```diff
@@ -0,0 +1,344 @@
+
+    /**
+     * m:n relationship
+     */
+    describeIf(isMongoDB)('m:n mandatory (explicit) - MongoDB', () => {
+      const postModel = 'PostManyToMany'
+      const categoryModel = 'CategoryManyToMany'
+
+      beforeEach(async () => {
+        const prismaPromises = [prisma[postModel].deleteMany(), prisma[categoryModel].deleteMany()]
+        await prisma.$transaction(prismaPromises)
+      })
+
+      describe('[create]', () => {
+        // Only this test and the one below pass
+        test('[create] category alone should succeed', async () => {
+          await prisma[categoryModel].create({
+            data: {
+              id: '1',
+            },
+          })
+          expect(await prisma[categoryModel].findMany()).toEqual([
+            {
+              id: '1',
+              published: null,
+              postIDs: [],
+            },
+          ])
+        })
+
+        // Only this test and the one above pass
+        test('[create] post alone should succeed', async () => {
+          await prisma[postModel].create({
+            data: {
+              id: '1',
+            },
+          })
+          expect(await prisma[postModel].findMany()).toEqual([
+            {
+              id: '1',
+              published: null,
+              categoryIDs: [],
+            },
+          ])
+        })
+
+        test.failing(
+          '[create] create post [nested] [create] categories [nested] [create] category should succeed',
+          async () => {
+            await prisma[postModel].create({
+              data: {
+                id: '1',
+                categories: {
+                  create: [
+                    {
+                      id: '1',
+                    },
+                  ],
+                },
+              },
+            })
+
+            expect(await prisma[postModel].findMany({ orderBy: { id: 'asc' } })).toEqual([
+              {
+                id: '1',
+                published: null,
+                categoryIDs: ['1'],
+              },
+            ])
+            expect(await prisma[categoryModel].findMany()).toEqual([
+              {
+                id: '1',
+                published: null,
+                postIDs: ['1'],
+              },
+            ])
+          },
+        )
+      })
+
+      describe('[update]', () => {
+        beforeEach(async () => {
+          await checkIfEmpty(categoryModel, postModel)
+          await createXPostsWith2CategoriesMongoDB({
+            count: 2,
+            postModel,
+          })
+        })
+
+        // Note: it's not possible on MongoDB to mutate _id, it is immutable
+        test.failing('[update] id (_id) should throw at runtime because id field is read-only/immutable', async () => {
+          await expect(
+            prisma[postModel].update({
+              where: {
+                id: '1',
+              },
+              data: {
+                // This would show a type error
```

Comment:
We have magic comments like for conditional type checks
`// @ts-test-if: provider !== 'mongodb'`

Reply:
But we don't ignore it here, so adding this condition won't change anything?

Reply:
This code is only running for mongodb here